Makefile

```diff
@@ -125,7 +125,7 @@ $(BIN_DIR)/azure-acr-credential-provider.exe: $(PKG_CONFIG) $(wildcard cmd/acr-c
 .PHONY: docker-pull-prerequisites
 docker-pull-prerequisites: ## Pull prerequisite images.
 	docker pull docker/dockerfile:1
-	docker pull docker.io/library/golang:buster
+	docker pull docker.io/library/golang:bullseye
```

Comment:
i am not sure why it's being pre-pulled here, but if you are prepulling I would set this a a variable instead of defining in multiple places, and then pass in as docker build args. However, dependabot won't be able to update the image.

Reply:
Also not sure, just trying to get this update process started. @feiskyer @lzhecheng feel free to offer a better path forward that doesn't require refactoring all of the existing CI.

Reply:
It doesn't seem necessary to me. Can we get rid of this target? @feiskyer

Reply:
@lodrem can we remove this target?

Reply:
let's get this step removed.

Reply:
@jackfrancis confirmed with Pengfei and we think this target can be removed. Could you please update the PR and rebase? Thank you!

Reply:
Rebased and removed `docker-pull-prerequisites` entirely. Let me know if that's not what you meant!

Reply:
Thank you!
There's a merge conflict and a typo (not from this PR) blocking this PR from merging. Since the change is small, I pushed by myself to this PR.